src/index.d.ts

```diff
@@ -14,6 +14,30 @@ declare module 'bootstrap-vue' {
 
   export default VuePlugin
 
+  type VueVariant = 'active' | 'success' | 'info' | 'warning' | 'danger'
+  type SortDirection = 'asc' | 'desc'
+  type FormatterCallback = ((value: mixed, key: string, item: mixed) => string)
+
+  export interface TableField {
+    label?: string
+    headerTitle?: string
+    headerAbbr?: string
+    class?: string | string[]
+    formatter?: string | FormatterCallback
+    sortable: boolean
+    sortDirection: SortDirection
+    tdClass: string | string[] | FormatterCallback
+    thClass: string | string[]
+    thStyle: mixed
+    variant: VueVariant
```

Comment:
Different components have different variants available to them... so I don't think this type should be called `VueVariant`... but maybe `TableVariant` (and even then, people can defined their own variants to use, so maybe this should be `string`, or add the ability to use the defined variants, plus a catch all string to say it could be custom)